File: python/examples/svr_regression.py
from sg import sg
from pylab import figure,plot,show
from numpy import array,transpose,sin,double

sg('send_command', 'new_svm LIBSVR')
features=array([range(0,100)],dtype=double)
features.resize(1,100)
labels=sin(features)[0]
sg('set_features', "TRAIN", features)
sg('set_labels', "TRAIN", labels)
sg('send_command', 'set_kernel GAUSSIAN REAL 20 10')
sg('send_command', 'init_kernel TRAIN')
sg('send_command', 'c 1')
sg('send_command', 'svm_train')
[bias, alphas]=sg('get_svm');
sg('set_features', "TEST", features)
sg('send_command', 'init_kernel TEST')
out=sg('svm_classify');

figure()
plot(features[0],labels,'b-')
plot(features[0],labels,'bo')
plot(features[0],out,'r-')
plot(features[0],out,'ro')
show()

File: matlab_and_octave/examples/plots/svr_regression.m
traindat = (1:100)/10;
trainlab = sin(traindat);
testdat = ((1:100)-0.5)/10;
testlab = sin(testdat);

sg('set_features', 'TRAIN', traindat);
sg('set_labels', 'TRAIN', trainlab);
sg('send_command', 'set_kernel GAUSSIAN REAL 10 0.1');
sg('send_command', 'init_kernel TRAIN');
sg('send_command', 'new_svm LIBSVR');
sg('send_command', 'c 0.1');
sg('send_command', 'svr_tube_epsilon 0.2');
sg('send_command', 'svm_train');
sg('set_features', 'TEST', testdat);
sg('set_labels', 'TEST', testlab);
sg('send_command', 'init_kernel TEST');
out=sg('svm_classify');
err=mean((testlab-out).^2);

figure(1)
clf
plot(traindat, trainlab,'bo-')
hold on
plot(testdat, testlab,'ro-')
